gammapy/background/models.py

```diff
@@ -71,3 +82,724 @@ def evaluate(self, x, y):
+
+        # check units
+        if (im_header['CUNIT1'] != im_header['CUNIT2']):
+            ss_error = "This is odd: detector X and Y units not matching"
+            ss_error += "({0}, {1})".format(im_header['CUNIT1'], im_header['CUNIT2'])
+            raise ValueError(ss_error)
+        if (im_header['CUNIT3'] != en_header['TUNIT1']):
+            ss_error = "This is odd: energy units not matching"
+            ss_error += "({0}, {1})".format(im_header['CUNIT3'], en_header['TUNIT1'])
+            raise ValueError(ss_error)
+
+        # get det X, Y binning
+        wcs = WCS(im_header, naxis=2) # select only the (X, Y) axes
+        detx_bins, dety_bins = linear_arrays_from_wcs(wcs,
+                                                      im_header['NAXIS1'],
+                                                      im_header['NAXIS2'])
+
+        # get energy binning
+        energy_bins = Quantity(enhdu.data['ENERGY_BIN_EDGES'],
+                               en_header['TUNIT1'])
+
+        # get background data
+        background = imhdu.data
+        background_unit = _parse_bg_units(im_header['BG_UNIT'])
+        background = Quantity(background, background_unit)
+
+        return CubeBackgroundModel(detx_bins=detx_bins,
+                                   dety_bins=dety_bins,
+                                   energy_bins=energy_bins,
+                                   background=background)
+
+    @staticmethod
+    def read(filename, format='table'):
+        """Read cube background model from fits file.
+
+        Several input formats are accepted, depending on the value
+        of the `format` parameter:
+
+        * bin_table (default and preferred format): `~astropy.io.fits.BinTableHDU`
+
+        * image (alternative format): `~astropy.io.fits.ImageHDU`,
+          with the energy binning stored as `~astropy.io.fits.BinTableHDU`
+
+        Parameters
+        ----------
+        filename : `~str`
+            name of file with the bg cube
+        format : `~str`, optional
+            format of the bg cube to read
+
+        Returns
+        -------
+        bg_cube : `~gammapy.background.CubeBackgroundModel`
+            bg model cube object
+        """
+        hdu = fits.open(filename)
+        if format == 'table':
+            return CubeBackgroundModel.from_fits_bin_table(hdu['BACKGROUND'])
+        elif format == 'image':
+            return CubeBackgroundModel.from_fits_image(hdu['BACKGROUND'], hdu['ENERGY_BINS'])
+        else:
+            raise ValueError("Invalid format {}.".format(format))
+
+    def to_table(self):
+        """Convert cube background model to astropy table format.
+
+        The name of the table is stored in the table meta information
+        under the keyword 'name'.
+
+        Returns
+        -------
+        table : `~astropy.table.Table`
+            table containing the bg cube
+        """
+        # data arrays
+        a_detx_lo = Quantity([self.detx_bins[:-1]])
```

Comment:
So you basically want to reshape the array here?
Then please call reshape or maybe [numpy.expand_dims](http://docs.scipy.org/doc/numpy/reference/generated/numpy.expand_dims.html) to make the code clearer.
(I hope it works with Quantity objects).

Also, in this case the temp variables aren't really needed, you can just assign to the table columns below and it'll still nicely fit on one line, no?